lib/carrierwave/storage/app.rb

```diff
@@ -0,0 +1,78 @@
+require 'zip'
+
+module CarrierWave
+  module Storage
+    class App < Abstract
+      def store!(file)
+        tmp_dir = Dir.mktmpdir('plgapp')
+        Zip::File.open(file.file) do |zipfile|
+          zipfile.each do |f|
+            dest_file = ::File.expand_path(f.name, tmp_dir)
+            f.extract(dest_file)
+          end
+
+          FileUtils.rm_r(app_dir)
+          FileUtils.mv(tmp_dir, app_dir)
+        end
+      end
+
+      def retrieve!(identifier)
```

Comment:
Unused method argument - `identifier`. If it's necessary, use `_` or `_identifier` as an argument name to indicate that it won't be used. You can also write as `retrieve!(*)` if you want the method to accept any arguments but don't care about them.